src/interface_adapter/AccountCreation/AccountCreationPresenter.java

```diff
@@ -20,16 +20,22 @@ public void setPassView(AccountCreationOutputData response) {
         AccountCreationState accountCreationState = accountCreationViewModel.getState();
         accountCreationState.setUsernameError(response.getUsername());
         accountCreationState.setPasswordError(null);
-        accountCreationViewModel.firePropertyChanged("success", null, "Account Created!");
+        accountCreationViewModel.firePropertyChanged("success");
     }
 
     @Override
     public void setFailView(String error) {
-//        AccountCreationState accountCreationState = accountCreationViewModel.getState();
-//        accountCreationState.setUsernameError(error);
-//        //TODO: Set correct error and display an error screen.
-//        accountCreationViewModel.setState(accountCreationState);
-//        accountCreationViewModel.firePropertyChanged(error);
-        System.out.println("Error");
+        AccountCreationState accountCreationState = accountCreationViewModel.getState();
+        accountCreationState.setErrorCode(error);
+        if (error.equals("unmatched_passwords")) {
+            accountCreationState.setErrorCode("Passwords do not match.");
+        }
+        if (error.equals("missing_fields")) {
+            accountCreationState.setErrorCode("Please fill in all fields.");
```

Comment:
setErrorCode should have another variable name. This one is not descriptive enough

Reply:
You're right, I'll refactor this variable name to be more indicative of what it serves. Thanks for the feedback and review.